File: constructor.dart
class Employee {
  int empId = 1;
  String empName = "Saraswathi";

//Parametrized constructor
  Employee(int empId, String empName) {
    print("Parameterized Constructor");
    this.empId = empId;
    this.empName = empName;
  }

  //Named Constructor
  Employee.myOwnCustomConstructor() {
    print("Named Constructor");
    this.empId = empId;
    this.empName = empName;
  }

  Employee.namedConstructor(this.empId, this.empName) {
    print("Another Named Constructor");
  }

  
  void display() {
    print("Employee id is: $empId");
    print("Employee name  is: $empName");
  }
}

void main() {
  Employee emp = new Employee(7, 'Prabha');
  emp.display();
  print("\n");
  Employee emp1 = Employee.myOwnCustomConstructor();
  emp1.empId = 12;
  emp1.empName = 'Sabana';
  emp1.display();
  print("\n");
  Employee emp2 = Employee.namedConstructor(29, 'Thiru');
  emp2.display();
}

//Const Constructor
class Student {
  final String name;
  final int age;

  const Student(this.name, this.age);
}

void main() {
  Student s = const Student('Prabha', 22);
  print("Employee name  is: ${s.name}");
  print("Employee name  is: ${s.age}");
}

File: src/Constructor.java
public class Constructor extends ParentConst {
    public Constructor(){
        super(123);
        System.out.println("Default Constructor");

    }
    public Constructor(int id){
        super(50.50f);
        System.out.println("Integer Parameterized constructor");
        System.out.println("Student id:"+ id);
        System.out.println("\n");
    }
    public Constructor(String name,long phno){
        System.out.println("String and Long parametrized constr");
        System.out.println("Student name:"+name);
        System.out.println("Student Phone_No:"+phno);
        System.out.println("\n");
    }

    public static void main(String[] args) {
        Constructor c = new Constructor();
        Constructor c1 = new Constructor(10);
        Constructor c2 = new Constructor("Prabha",8756406213L);

    }
}
